src/client/java/teammates/client/scripts/sql/VerifyCourseEntityAttributes.java

```diff
@@ -154,14 +163,180 @@ private boolean verifyStudent(CourseStudent oldStudent,
+        List<teammates.storage.sqlentity.FeedbackSession> newSessions = newCourse.getFeedbackSessions();
+        List<FeedbackSession> oldSessions = ofy().load().type(FeedbackSession.class)
+                .filter("courseId", newCourse.getId()).list();
+
+        if (newSessions.size() != oldSessions.size()) {
+            log(String.format("Mismatched session counts for course id: %s", newCourse.getId()));
+            return false;
+        }
+
+        Map<String, FeedbackSession> sessionNameToOldSessionMap = oldSessions.stream()
+                .collect(Collectors.toMap(FeedbackSession::getFeedbackSessionName, session -> session));
+
+        return newSessions.stream().allMatch(newSession -> {
+            FeedbackSession oldSession = sessionNameToOldSessionMap.get(newSession.getName());
+            return verifyFeedbackSession(oldSession, newSession);
+        });
+    }
+
+    private boolean verifyFeedbackSession(FeedbackSession oldSession, teammates.storage.sqlentity.FeedbackSession newSession) {
+        boolean doFieldsMatch = newSession.getCourse().getId().equals(oldSession.getCourseId())
+                && newSession.getName().equals(oldSession.getFeedbackSessionName())
+                && newSession.getCreatorEmail().equals(oldSession.getCreatorEmail())
+                && newSession.getInstructions().equals(SanitizationHelper.sanitizeForRichText(oldSession.getInstructions()))
+                && newSession.getStartTime().equals(oldSession.getStartTime())
+                && newSession.getEndTime().equals(oldSession.getEndTime())
+                && newSession.getSessionVisibleFromTime().equals(oldSession.getSessionVisibleFromTime())
+                && newSession.getResultsVisibleFromTime().equals(oldSession.getResultsVisibleFromTime())
+                && newSession.getGracePeriod().equals(Duration.ofMinutes(oldSession.getGracePeriod()))
+                && newSession.isOpeningEmailEnabled() == oldSession.isOpeningEmailEnabled()
+                && newSession.isClosingEmailEnabled() == oldSession.isClosingEmailEnabled()
+                && newSession.isOpenEmailSent() == oldSession.isSentOpenEmail()
+                && newSession.isOpeningSoonEmailSent() == oldSession.isSentOpeningSoonEmail()
+                && newSession.isClosedEmailSent() == oldSession.isSentClosedEmail()
+                && newSession.isClosingSoonEmailSent() == oldSession.isSentClosingEmail()
+                && newSession.isPublishedEmailSent() == oldSession.isSentPublishedEmail()
+                && newSession.getCreatedAt().equals(oldSession.getCreatedTime())
+                && (newSession.getDeletedAt() == oldSession.getDeletedTime()
+                        || newSession.getDeletedAt().equals(oldSession.getDeletedTime()));
+        if (!doFieldsMatch) {
+            log(String.format("Mismatched fields for session: %s, course id: %s",
+                    oldSession.getFeedbackSessionName(), oldSession.getCourseId()));
+            return false;
```

Comment:
I'm thinking if we should instead continue to verify the rest of the chain to get all the errors before we fix them all at once. Only being able to see one error could make us repeatedly run the verification to fix one error at a time relating to this course

Reply:
I was thinking that we could deal with one issue at a time, instead of having to parse a giant wall of errors, since its probable that the initial issue causes many of the ensuing entities to fail. I feel that this approach might be more manageable for debugging, what do you think?

Reply:
Yea it occured to me that if there is an error, there might not be no point verifying the rest. We can leave it as such